### PR TITLE
Include Microsoft.NETCore.App.Ref pack as part of 6.0.2 release

### PR DESCRIPTION
**Pull request by safern**
There are a bunch of source generator fixes that ship with the ref pack so we need to build it for 6.0.2.

### CODE REVIEW

**Review by joperezr — APPROVED** [review]
LGTM. I suppose the SDK will automatically pick this up? Also, no need to add a GeneratePackageOnBuild here?

File: eng/Versions.props [hunk 1]
```diff
@@ -38,7 +38,7 @@
   -->
   <ItemGroup>
     <!-- Targeting packs are only patched in extreme cases. -->
-    <ProjectServicingConfiguration Include="Microsoft.NETCore.App.Ref" PatchVersion="1" />
+    <ProjectServicingConfiguration Include="Microsoft.NETCore.App.Ref" PatchVersion="2" />
```
Comment (ericstj) [inline eng/Versions.props:41]: We should change this comment. `Targeting packs are only patched in extreme cases`

How about `Targeting packs are only patched when we need to patch reference assemblies, intellisense, or source generators`

**Review by ericstj — APPROVED** [review]

File: eng/Versions.props (hunk 1 above)
Comment (safern) [reply eng/Versions.props:41]: I'll update it.

File: eng/Versions.props [hunk 2]
```diff
@@ -38,7 +38,7 @@
   -->
   <ItemGroup>
     <!-- Targeting packs are only patched in extreme cases. -->
```
Comment (safern) [inline eng/Versions.props:40]:
```suggestion
    <!-- Targeting packs are only patched when we need to patch reference assemblies, intellisense, or source generators. -->
```

**Review by ericstj — APPROVED** [review]

**Comment by safern** [comment]
> LGTM. I suppose the SDK will automatically pick this up? Also, no need to add a GeneratePackageOnBuild here?

This is handled by the shared framework SDK: 
https://github.com/dotnet/arcade/blob/e7ede87875f41a9b3df898ae08da5ebc96e24f56/src/Microsoft.DotNet.SharedFramework.Sdk/targets/Microsoft.DotNet.SharedFramework.Sdk.targets#L13

https://github.com/dotnet/arcade/blob/e7ede87875f41a9b3df898ae08da5ebc96e24f56/src/Microsoft.DotNet.SharedFramework.Sdk/targets/Microsoft.DotNet.SharedFramework.Sdk.targets#L56

**Pull request merged**